packages/e2e-test-utils/src/rest-api.js

```
@@ -12,14 +14,110 @@ import apiFetch from '@wordpress/api-fetch';
 /**
  * Internal dependencies
  */
-import { WP_BASE_URL } from './shared/config';
+import { WP_BASE_URL, WP_ADMIN_USER } from './shared/config';
 import { createURL } from './create-url';
 
+/**
+ * @typedef {Object} StorageState
+ * @property {string} cookie  The login cookie.
+ * @property {string} nonce   The login nonce.
+ * @property {string} rootURL The REST API root url.
+ */
+
 // `apiFetch` expects `window.fetch` to be available in its default handler.
 global.window = global.window || {};
 global.window.fetch = fetch;
 
-const setAPIRootURL = ( async () => {
+const REST_NONCE_ENDPOINT = createURL(
+	'wp-admin/admin-ajax.php',
+	'action=rest-nonce'
+);
+
+class AdminStorageState {
+	/**
+	 * Initialize the admin storage state from disk.
+	 *
+	 * @param {string} [storageStatePath] The storage state path.
+	 * @return {Promise<AdminStorageState>} The AdminStorageState instance.
+	 */
+	static async init( storageStatePath ) {
+		/** @type {Partial<StorageState>} */
+		let initialStorageState = {};
+
+		if ( storageStatePath ) {
+			try {
+				initialStorageState = JSON.parse(
+					await fs.readFile( storageStatePath, 'utf-8' )
+				);
+			} catch ( error ) {
+				// Ignore errors if the state is not found or invalid.
```

File not found seems like a reasonable thing to ignore, but invalid seems problematic and perhaps this should throw an error.